### PR TITLE
fix: Do not add invalid Cisco senors seen in IOS 15.6(3)M1

### PR DESCRIPTION
**Pull request by Rosiak**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

Cisco introduced invalid sensors into IOS 15.6(3)M1, this fix makes sure we do not add these sensors.

### CODE REVIEW

File: includes/discovery/sensors/state/cisco.inc.php [hunk 1]
```diff
@@ -135,26 +136,31 @@
...
-                $descr = $tablevalue[3] . $swrolenumber;
-            } elseif ($state_name == 'cswSwitchState') {
-                $swstatenumber++;
-                $descr = $tablevalue[3] . $swstatenumber;
-            } elseif ($state_name == 'cswStackPortOperStatus') {
-                $stack_port_descr = get_port_by_index_cache($device, $index);
-                $descr = $tablevalue[3] . $stack_port_descr['ifDescr'];
-            } elseif ($state_name == 'cefcFRUPowerOperStatus') {
-                $descr = snmp_get($device, 'entPhysicalName.'.$index, '-Oqv', 'ENTITY-MIB');
+            if ($tablevalue[2] == 'ciscoEnvMonTemperatureState' && (is_null($temp[$index][$tablevalue[3]]) or empty($temp[$index][$tablevalue[3]]))) {
```
Comment (laf) [inline includes/discovery/sensors/state/cisco.inc.php]: `or` should be a `||` to be consistent.

Is `empty()` not enough here or is `is_null()` needed as well?

File: includes/discovery/sensors/state/cisco.inc.php [hunk 2]
```diff
@@ -135,26 +136,31 @@
...
+                $process_sensor = false;
             }
-            discover_sensor($valid['sensor'], 'state', $device, $cur_oid.$index, $index, $state_name, $descr, '1', '1', null, null, null, null, $temp[$index][$tablevalue[2]], 'snmp', $index);
+            if ($process_sensor == true) {
```
Comment (laf) [inline includes/discovery/sensors/state/cisco.inc.php]: Why not just leave this to the previous `if()` check that I commented on, seems pointless setting a variable.

File: includes/discovery/sensors/state/cisco.inc.php (hunk 1 above)
Comment (murrant) [reply includes/discovery/sensors/state/cisco.inc.php]: Yeah, just empty should be fine: http://php.net/manual/en/types.comparisons.php

File: includes/discovery/sensors/state/cisco.inc.php (hunk 2 above)
Comment (Rosiak) [reply includes/discovery/sensors/state/cisco.inc.php]: Not sure what you mean here.
What are you suggesting?

Comment (laf) [reply includes/discovery/sensors/state/cisco.inc.php]: Litterally just change it from:

```php
}
if ($process_sensor == true) {
```

To:

```php
} else {
```

No point in assigning a variable in one if statement only to open a new one using that variable.

**Comment by laf** [comment]
Does an empty description definitely mean bad sensor?

File: includes/discovery/sensors/state/cisco.inc.php (hunk 2 above)
Comment (Rosiak) [reply includes/discovery/sensors/state/cisco.inc.php]: Now I understand what you mean, you are totally right.

**Comment by Rosiak** [comment]
@laf I won't say I'm 100% sure, but I've only seen it empty in this software version so far.

**Pull request merged**